pkg/controller/attach_detach.go

```diff
@@ -312,64 +314,60 @@ func (r *ReconcileAttachDetach) triggerDetach(ctx context.Context, azVolumeAttac
 	var err error
 	ctx, w := workflow.New(ctx)
 	defer func() { w.Finish(err) }()
-	// only detach if detachment request was made for underlying volume attachment object
-	detachmentRequested := volumeDetachRequested(azVolumeAttachment)
 
-	if detachmentRequested {
-		defer r.stateLock.Delete(azVolumeAttachment.Name)
-		if _, ok := r.stateLock.LoadOrStore(azVolumeAttachment.Name, nil); ok {
-			return getOperationRequeueError("detach", azVolumeAttachment)
-		}
+	defer r.stateLock.Delete(azVolumeAttachment.Name)
```

Comment:
Should the delete of the azva state lock occur only if we acquire the lock on the next line? Otherwise, we'll delete the state lock when we haven't acquired it.

Reply:
Yeah, it makes sense to me. I'll change all of them in our codebase.